client/app/services/designer-state.service.js

```diff
@@ -141,12 +105,19 @@
 
       function matchAtomicServiceItemToTabs(srvTemplate) {
         var subTab;
-        var newItem;
         if (srvTemplate.service_type === 'atomic') {
           if (srvTemplate.prov_type !== 'generic') {
             subTab = findSubTabByProvType(srvTemplate.prov_type);
-          } else {
+            if (!subTab) {
+              console.log("Couldn't find toolbox Tab for Prov_Type: " + srvTemplate.prov_type);
```

Comment:
oh great point, something might be up if indeeeeed its turned on...` angular/log` is the error we'll get, using `console.log` is bad form